getting_serious/puzzle_29.R

```diff
@@ -0,0 +1,47 @@
+## Ten people, two rooms, states are {0, 1, 2, ..., 10} (number of people in other room)
+states <- seq(0, 10)
+
+## Transition matrix P
+## P[i, j] is the probability of transitioning from state i to j
+P <- matrix(0, length(states), length(states))
+rownames(P) <- states
+colnames(P) <- states
+for(i in seq_along(states)[seq(2, length(states) - 1)]) {
+    P[i, i-1] <- (i-1) / max(states)  # Person in other room selected, state decreases
```

Comment:
This line is confusing because the two instances of `i-1` mean completely different things, yet they look the same.